fuse_optimizers/src/variable_stamp_index.cpp

```diff
@@ -99,6 +109,10 @@ void VariableStampIndex::applyAddedVariables(const fuse_core::Transaction& trans
     if (stamped_variable)
     {
       stamped_index_[variable.uuid()] = stamped_variable->stamp();
+      // add to unique stamps
+      if(unique_stamps_.find(stamped_variable->stamp()) == unique_stamps_.end()){
+        unique_stamps_.insert(stamped_variable->stamp());
+      }
```

Comment:
I don't think there is a need to query of rht e stamp first. If it already exists, `insert()` will not insert it again.
```suggestion
      unique_stamps_.insert(stamped_variable->stamp());
```